utils/getAllMarkdownFiles.ts

```
@@ -0,0 +1,47 @@
+// utils/getAllMarkdownFiles.ts
+
+import * as fs from 'fs';
+import * as path from 'path';
+
+interface getAllMarkdownFilesOptions {
+    /**
+     * List of filenames (case-insensitive) to skip/ignore.
+     * e.g. ['index.md']
+     */
+    ignoreFilenames?: string[]
+}
+
+
+export function getAllMarkdownFiles(
+    dirPath: string,
```

If you're given an invalid directory path, silently returning an empty array is misleading. Should throw an error instead!

---

I agree. Improved the error handling.